gh-56276: Add tests to test_compare

csabella opened the pull request:
The part of the Maier-Panter bpo patch that hasn't been committed yet (doc changes were).

Contributed by: Andy Maier.
Contributed by: Martin Panter.

<!-- gh-issue-number: gh-56276 -->
* Issue: gh-56276
<!-- /gh-issue-number -->

csabella committed:
bpo-12067: Add tests to test_compare

terryjreedy commented on Lib/test/test_compare.py:
```diff
@@ -1,27 +1,40 @@
+
+
 import unittest
+from fractions import Fraction
```
```suggestion
from test.support import ALWAYS_EQ
from fractions import Fraction
```

terryjreedy committed:
From test_compare part of PR-15167.

From test_compare part of PR-15167.

terryjreedy commented on Lib/test/test_compare.py:
```diff
@@ -1,27 +1,41 @@
-        return True
+    class Anything:
+        def __eq__(self, other):
+            return True
```
```suggestion
```

terryjreedy committed:
@terryjreedy From test_compare part of PR-15167.

terryjreedy commented on Lib/test/test_compare.py:
```diff
@@ -1,27 +1,41 @@
-    def __ne__(self, other):
-        return False
+        def __ne__(self, other):
+            return False
```
```suggestion
```

terryjreedy committed:
@terryjreedy From test_compare part of PR-15167.

terryjreedy commented on Lib/test/test_compare.py:
```diff
@@ -113,11 +121,614 @@ class C:
-        self.assertEqual(x, Anything())
-        self.assertEqual(Anything(), x)
+        self.assertEqual(x, self.Anything())
+        self.assertEqual(self.Anything(), x)
```
```suggestion
        self.assertEqual(x, ALWAYS_EQ)
        self.assertEqual(ALWAYS_EQ, x)
```

terryjreedy commented on Lib/test/test_compare.py:
```diff
@@ -113,11 +121,614 @@ class C:
-        self.assertEqual(y, Anything())
-        self.assertEqual(Anything(), y)
+        self.assertEqual(y, self.Anything())
+        self.assertEqual(self.Anything(), y)
```
```suggestion
        self.assertEqual(y, ALWAYS_EQ)
        self.assertEqual(ALWAYS_EQ, y)
```

terryjreedy committed:
@terryjreedy From test_compare part of PR-15167.

terryjreedy committed:
@terryjreedy From test_compare part of PR-15167.

terryjreedy committed:
Merge branch 'main' into compare

arhadthedev committed:
Merge branch 'main' into compare

terryjreedy committed:
Merge branch 'main' into compare

terryjreedy committed:
Replace 'insts' with 'instances'.

terryjreedy committed:
Factor repeated comments out of comparison subtests.

terryjreedy committed:
Edit/condense docstrings/comments; split test_comp_classes.

terryjreedy committed:
Remove identity claims and tests within test_str_subclass.
They passed because they remain correct for CPython and the
toy subclass, but are not part of the definition of Python.

terryjreedy committed:
Edit more docstrings, remove duplicate ints.

terryjreedy committed:
Edit create_sorted_instances.

terryjreedy committed:
Edit assert_total_order and subtests.

terryjreedy committed:
Condense redundant subtest docstrings.

terryjreedy committed:
Finish editing test_numbers.

terryjreedy committed:
Fix test_sequences by removing duplicate objects and 'is' tests.

terryjreedy committed:
Fix test_bytes like test_sequences.

terryjreedy committed:
More docstrings.

terryjreedy committed:
Edit test_sets and test_mappings.

arhadthedev commented on Lib/test/test_compare.py:
```diff
@@ -1,21 +1,27 @@
+"Test equality and order comparisons."
```
Do docstrings support single quotes?

```suggestion
"""Test equality and order comparisons."""
```

terryjreedy replied:
Docstrings are string literals (not f-strings) that are the first statement in a module, class, or function body.  The parser does not know what quotes and prefixes (other than f) were removed by the tokenizer.  Using `"` rather than `'` is simply a style convention.  I personally use `"` rather than `"""` for single line docstrings, but since the author of the module and the patch did otherwise elsewhere, I will switch this one.

arhadthedev commented on Lib/test/test_compare.py:
```diff
@@ -114,5 +119,391 @@ def test_issue_1393(self):
+        # if value-based comparison methods:
+            # expect what the testcase defined for a op b and b rop a;
+        # else:  no value-based comparison
+            # expect default behavior of object for a op b and b rop a.
```
Maybe turn this into a single block?

```suggestion
    # The body of each subtest has form:
    #
    #     if value-based comparison methods:
    #         expect what the testcase defined for a op b and b rop a;
    #     else:  no value-based comparison
    #         expect default behavior of object for a op b and b rop a.
```

terryjreedy replied:
I agree it looks better.

terryjreedy committed:
Update Lib/test/test_compare.py

Co-authored-by: Oleg Iarygin <oleg@arhadthedev.net>

terryjreedy committed:
Update Lib/test/test_compare.py

Co-authored-by: Oleg Iarygin <oleg@arhadthedev.net>

terryjreedy committed:
Merge branch 'main' into compare
